python/lsst/daf/butler/core/composites.py

```diff
@@ -102,12 +103,11 @@ def doDisassembly(self, entity):
         matchName = "{} (via default)".format(entity)
         disassemble = self.config["default"]
 
-        if datasetTypeName is not None and datasetTypeName in self.config["datasetTypes"]:
-            disassemble = self.config[f"datasetTypes.{datasetTypeName}"]
-            matchName = datasetTypeName
-        elif storageClassName is not None and storageClassName in self.config["storageClasses"]:
-            disassemble = self.config[f"storageClasses.{storageClassName}"]
-            matchName = storageClassName
+        for name in (datasetTypeName, storageClassName):
+            if name is not None and name in self.config["names"]:
+                disassemble = self.config[f"names.{name}"]
+                matchName = name
+                break
```

Comment:
So we better hope that there is never a clash between a storageClassName and a datasetTypeName (the latter being something that we can't control and rely on users not to pick incorrectly)?

Reply:
Yes. We have to decide. At the moment we decided that they can't clash. If we want to ensure they can't clash we need to decide which of the options we want to use (as defined in my previous comment: separate sections in every config, or insist on naming convention).